src/main/java/ch/njol/skript/expressions/ExprLore.java

```diff
@@ -1,236 +1,181 @@
-		if (meta == null || !meta.hasLore())
-			return new String[0];
-		final List<String> lore = meta.getLore();
-		assert lore != null; // hasLore() called before
-		if (n == null)
-			return lore.toArray(new String[0]);
-		final int l = n.intValue() - 1;
-		if (l < 0 || l >= lore.size())
-			return new String[0];
-		return new String[]{lore.get(l)};
-	}
 
-	@Override
-	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		boolean acceptsMany = lineNumber == null;
-		switch (mode) {
-			case REMOVE:
-			case REMOVE_ALL:
-			case DELETE:
-				acceptsMany = false;
-			case SET:
-			case ADD:
-				if (ChangerUtils.acceptsChange(item, ChangeMode.SET, ItemStack.class, ItemType.class)) {
-					return CollectionUtils.array(acceptsMany ? String[].class : String.class);
-				}
-				return null;
-			case RESET:
-			default:
-				return null;
+		if (lineNumber == null) {
+			return itemLore.toArray(String[]::new);
+		}
+
+		int loreIndex = this.lineNumber.getOptionalSingle(event).orElse(0).intValue() -1;
+		if (loreIndex < 0 || loreIndex >= itemLore.size()) {
+			return null;
 		}
+		return new String[]{itemLore.get(loreIndex)};
 	}
 
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) throws UnsupportedOperationException {
-		Object i = item.getSingle(e);
-
-		String[] stringDelta = delta == null ? null : Arrays.copyOf(delta, delta.length, String[].class);
-
-		// air is just nothing, it can't have a lore
-		if (i == null || i instanceof ItemStack && ((ItemStack) i).getType() == Material.AIR)
-			return;
-
-		ItemMeta meta = i instanceof ItemStack ? ((ItemStack) i).getItemMeta() : (ItemMeta) ((ItemType) i).getItemMeta();
-		if (meta == null)
-			meta = Bukkit.getItemFactory().getItemMeta(Material.STONE);
-
-		Number lineNumber = this.lineNumber != null ? this.lineNumber.getSingle(e) : null;
-		List<String> lore = meta.hasLore() ? new ArrayList<>(meta.getLore()) : new ArrayList<>();
-
-		if (lineNumber == null) {
-			// if the condition below is true, the pattern with the line %number% expression was used,
-			// but the line number turned out to be null at runtime, meaning we should ignore it
-			if (this.lineNumber != null) {
-				return;
+	public Class<?> @Nullable [] acceptChange(ChangeMode mode) {
+		if (this.item.getTime() != EventValues.TIME_NOW && !ChangerUtils.acceptsChange(item, ChangeMode.SET, ItemStack.class, ItemType.class)) {
+			// Allows past/future expressions only if they accept the set change mode
+			return null;
+		}
+		boolean acceptsMany = this.lineNumber == null;
+		return switch (mode) {
+			case SET -> CollectionUtils.array(acceptsMany ? String[].class : String.class);
+			case DELETE -> CollectionUtils.array();
+			case ADD, REMOVE, REMOVE_ALL -> {
+				if (!acceptsMany) {
+					Skript.error("You cannot add/remove the lore from a single line, you can however replace/concat lore.");
```

Comment:
```suggestion
					Skript.error("You cannot add to or remove from the lore from a single line.");
```
I think this is better for clarity (let me know if you disagree though)

Reply:
I was concerned about poor wording on my end, so this is a better alternative, just wasn't sure if we wanted to mention replace as an alternative, however since add said the same it might be better to not include it.